src/vs/platform/externalTerminal/node/externalTerminalService.ts

```diff
@@ -369,3 +375,152 @@ function quote(args: string[]): string {
+		cmdline: string[];
+		cwd: string;
+		pid: number;
+	}
+
+	export interface KittyWindow {
+		id: number;
+		title: string;
+		at_prompt: boolean;
+		foreground_processes: KittyProcess[];
+	}
+
+	export interface KittyTab {
+		windows: KittyWindow[];
+	}
+
+	export interface KittyInstance {
+		tabs: KittyTab[];
+	}
+}
+
+class KittyTerminalService {
+	static readonly KITTY_SOCKET = '/tmp/vscode-kitty.sock';
+	static readonly INVALID_WINDOW_ID = -1;
+
+	static async ensureKittyRunning() {
+		if (!existsSync(this.KITTY_SOCKET) || !statSync(this.KITTY_SOCKET).isSocket()) {
+			const child = cp.spawn('kitty', [
+				'-o', 'allow_remote_control=yes',
+				`--listen-on=unix:${this.KITTY_SOCKET}`
+			], {
+				detached: true,
+				stdio: 'ignore'
+			});
+			child.on('error', () => {
+				throw (new Error(nls.localize('mac.terminal.kitty.not.found', "Kitty terminal is not found.")));
+			});
+			child.unref(); // Fully detach the process
+
+			// Wait for the socket to be created
+			for (let i = 0; i < 10; i++) { // Retry for ~5 seconds
+				await delay(500);
+				if (existsSync(this.KITTY_SOCKET) && statSync(this.KITTY_SOCKET).isSocket()) {
+					return;
+				}
+			}
+			throw (new Error(nls.localize('mac.terminal.kitty.not.running', "Kitty terminal is not running")));
+		}
+	}
+
+	private static getInstances(): KittyTerminalService.KittyInstance[] {
+		try {
+			const result = cp.spawnSync('kitten', ['@', '--to', `unix:${this.KITTY_SOCKET}`, 'ls'], {
+				stdio: 'pipe',
+			});
+			if (result.status !== 0) {
+				const errorOutput = result.stderr.toString().trim() || 'Unknown error';
+				throw new Error(nls.localize('mac.terminal.kitty.list.failed', "Failed to list kitty windows: {0}", errorOutput));
+			}
+			const output = result.stdout.toString();
+			const instances = JSON.parse(output);
+			return instances;
+		} catch (error) {
+			throw new Error(nls.localize('mac.terminal.kitty.list.failed', "Failed to list kitty windows: {0}", error.message));
+		}
+	}
+
+	private static findAvailablePromptWindow(instances: KittyTerminalService.KittyInstance[], title: string): number {
+		for (const instance of instances) {
+			for (const tab of instance.tabs) {
+				for (const window of tab.windows) {
+					if (window.title.includes(title) && window.at_prompt) {
+						return window.id;
+					}
+				}
+			}
+		}
+		return this.INVALID_WINDOW_ID;
+	}
+
+	private static selectWindow(windowId: number): number {
+		if (windowId === this.INVALID_WINDOW_ID) {
+			return windowId;
+		}
+		const result = cp.spawnSync('kitten', ['@', '--to', `unix:${this.KITTY_SOCKET}`, 'select-window', '--match', `id:${windowId}`], {
+			stdio: 'pipe'
+		});
+		const selectedWindowId = result.status === 0 ? windowId : this.INVALID_WINDOW_ID;
+		return selectedWindowId;
+	}
+
+	private static launchPromptWindow(title: string): number {
+		const result = cp.spawnSync('kitten', ['@', '--to', `unix:${this.KITTY_SOCKET}`, 'launch', '--type', 'os-window', '--title', title], {
```

Comment:
We shouldn't be doing sync anywhere, it blocks the thread which would pause all lanugage services, extensions, etc.